iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Network.java

```diff
@@ -1199,14 +1303,134 @@ default int getConnectableCount() {
     BusView getBusView();
 
     /**
-     * Merge with an other network. At the end of the merge the other network
-     * is empty.
+     * Create an empty subnetwork in the current network.
+     *
+     * @param subnetworkId id of the subnetwork
+     * @param sourceFormat source format
+     * @return the created subnetwork
+     */
+    Network createSubnetwork(String subnetworkId, String sourceFormat);
+
+    /**
+     * Merge with another network. At the end of the merge the other network
+     * is empty (destructive merge).
      * @param other the other network
      */
     void merge(Network other);
 
     void merge(Network... others);
 
+    /**
+     * Unmerge the network. Each subnetwork will be converted in a fully-independent network.<br/>
+     * Each equipment linking 2 subnetworks together shall be removed prior to this operation.
+     * A {@link PowsyblException} will be thrown if 2 subnetworks are still linked together.<br/>
+     * If substations were defined at the root network level, the root network (without the subnetwork's equipments) will
+     * also be added to the resulting map. An {@link IllegalStateException} will be thrown if some links exist between
+     * these substations and another subnetwork.<br/>
+     *
+     * @return a map containing a new {@link Network} for each subnetwork detected ({@literal value}),
+     * associated to its id ({@literal key}).
+     */
+    Map<String, Network> unmerge();
+
+    /**
+     * Extract a subnetwork and convert it in a fully-independent network.<br/>
+     * Each equipment linking the subnetwork to another shall be removed prior to this operation.
+     * A {@link PowsyblException} will be thrown if the subnetwork is still linked to another one.<br/>
+     * All the extracted subnetworks' equipments will be removed from the current network.
+     *
+     * @param subnetworkId the id of a current network's subnetwork.
+     * @return a fully-independent {@link Network} corresponding to the subnetwork.
+     */
+    Network extractSubnetwork(String subnetworkId);
+
+    /**
+     * Check if the given identifiable id <b>directly</b> in the current network.<br/>
+     * If the identifiable is in a subnetwork of the current network, this method will return <code>false</code>.
+     *
+     * @param identifiable the identifiable to check
+     * @return true if the given identifiable is <b>directly</b> in the network
+     */
+    default boolean containsDirectly(Identifiable<?> identifiable) {
```

Comment:
What is the purpose of all theses methods:  containsDirectly, contains, getLinksBetween ?
What is the use case for that?
Could it be in a external utility class?